File: TodoApi/Models/Appointment.cs
namespace TodoApi.Models
{
    public class Appointment
    {
        public long Id { get; set; } // Unique identifier for the appointment
        public long RequestId { get; set; } // Foreign key for the request associated with the appointment
        public long RoomId { get; set; } // Foreign key for the room where the appointment takes place
        public DateTime Date { get; set; } // Date of the appointment
        public string Status { get; set; } = null!; // Status of the appointment (e.g., Scheduled, Completed, Canceled)
       
        
    }
}

File: TodoApi/bulletproof-nodejs-ddd/src/Models/Appointment.ts
import mongoose, {  Schema } from 'mongoose';
import { ISurgeryAppointment } from "./Interfaces/SurgeryAppointmentInterface";

const AppointmentSchema: Schema<ISurgeryAppointment> = new mongoose.Schema(
{
    RequestId: { type: String, required: true, unique: false },
    RoomId: { type: String, required: true, unique: false },
    Date: { type: String, required: true, unique: false },
    Status: { type: String, required: true, unique: false },
}
);

const Appointment= mongoose.model<ISurgeryAppointment>('Appointment', AppointmentSchema);
export default Appointment;